pkg/ebpf/controlplane/processes.go

```diff
@@ -13,213 +12,221 @@ import (
-			TaskHash:          taskHash,
-			ParentHash:        parentHash,
-			LeaderHash:        leaderHash,
-			CmdPath:           cmdPath,
-			PathName:          pathName,
-			Dev:               dev,
-			Inode:             inode,
-			Ctime:             ctime,
-			InodeMode:         inodeMode,
-			InterpreterPath:   interPathName,
-			InterpreterDev:    interDev,
-			InterpreterInode:  interInode,
-			InterpreterCtime:  interCtime,
-			Interp:            interp,
-			StdinType:         stdinType,
-			StdinPath:         stdinPath,
-			InvokedFromKernel: invokedFromKernel,
-		},
-	)
+	execFeed.StdinType, err = parse.ArgVal[uint16](args, "stdin_type")
+	if err != nil {
+		return err
+	}
+	execFeed.StdinPath, err = parse.ArgVal[string](args, "stdin_path")
+	if err != nil {
+		return err
+	}
+	execFeed.InvokedFromKernel, err = parse.ArgVal[int32](args, "invoked_from_kernel")
+	if err != nil {
+		return err
+	}
+
+	return ctrl.processTree.FeedFromExec(execFeed)
 }
 
 func (ctrl *Controller) procTreeExitProcessor(args []trace.Argument) error {
-	var errs []error
-
 	if ctrl.processTree == nil {
 		return nil // process tree is disabled
 	}
 
+	// NOTE: Currently FeedFromExit is only using TaskHash and TimeStamp from the ExitFeed.
+	// So the other fields are commented out for now.
+	//
+	// TODO: Analyze if the other fields will be needed in the future.
+	var err error
+
+	// NOTE: override all the fields of the exitFeed, to avoid any previous data.
+	exitFeed := ctrl.processTree.GetExitFeedFromPool()
+	defer ctrl.processTree.PutExitFeedInPool(exitFeed)
+
 	// Process & Event identification arguments (won't exist for regular events)
-	timestamp, err := parse.ArgVal[uint64](args, "timestamp")
-	errs = append(errs, err)
-	taskHash, err := parse.ArgVal[uint32](args, "task_hash")
-	errs = append(errs, err)
-	parentHash, err := parse.ArgVal[uint32](args, "parent_hash")
-	errs = append(errs, err)
-	leaderHash, err := parse.ArgVal[uint32](args, "leader_hash")
-	errs = append(errs, err)
-
-	// Exit logic arguments
-	exitCode, err := parse.ArgVal[int64](args, "exit_code")
-	errs = append(errs, err)
-	groupExit, err := parse.ArgVal[bool](args, "process_group_exit")
-	errs = append(errs, err)
-
-	// Handle errors
-	for _, err := range errs {
-		if err != nil {
-			return err
-		}
-	}
-
-	return ctrl.processTree.FeedFromExit(
-		proctree.ExitFeed{
-			TimeStamp:  time.BootToEpochNS(timestamp), // time of exit is already a times)p
-			TaskHash:   taskHash,
-			ParentHash: parentHash,
-			LeaderHash: leaderHash,
-			ExitCode:   exitCode,
-			Group:      groupExit,
-		},
-	)
+	exitFeed.TimeStamp, err = parse.ArgVal[uint64](args, "timestamp")
+	if err != nil {
+		return err
+	}
+	// time of exit is already a timestamp)
+	exitFeed.TimeStamp = time.BootToEpochNS(exitFeed.TimeStamp)
+
+	exitFeed.TaskHash, err = parse.ArgVal[uint32](args, "task_hash")
+	if err != nil {
+		return err
+	}
+	// exitFeed.ParentHash, err = parse.ArgVal[uint32](args, "parent_hash")
```

Comment:
Any reason of the following fields to be disable? `parent_hash`, `leader_hash`, `exit_code` and `process_group_exit`.

The benchmark seems to use the following fields:
https://github.com/aquasecurity/tracee/blob/f6bb523d3c4ab86f731307c092f57cc88fdaf2a8/pkg/ebpf/controlplane/processes_bench_test.go#L96-L103

Reply:
They are currently not used.

Reply:
Sorry, I conflated your initial question. They are commented out because of: https://github.com/aquasecurity/tracee/pull/4503/files/f6bb523d3c4ab86f731307c092f57cc88fdaf2a8#diff-cd791d8a1f8dc16e3ac078e7354a34a0d63c19fbe8254b1a283c63a937823382R190-R193.

The explanation is above. Regarding removing them from bench, I suppose it's different from interpreter ones, as they remain in the field but we avoid assign them since they are not used in the Feed, but there's a doubt about their requirement. Interpreter were not used at all, so we got them removed.